pkg/healthcheck/healthcheck.go

```diff
@@ -1215,6 +1218,15 @@ func (hc *HealthChecker) allCategories() []Category {
 						return
 					},
 				},
+				{
+					description: "can initialize the destination client",
```

Comment:
I'm not a big fan of adding this check, just so that we can generate a destination client as a side effect.  I think this means that this check would run as part of `linkerd check` which seems unnecessary (being able to create a destination client is not something you normally need) and also briefly creates an additional port-forward.

Can we remove this check and instead just rely on `hc.KubeAPIClient()` in `GetDestinationClient` to build the destination client explicitly, instead of doing it as a side-effect of a check?

Reply:
Makes sense, will push that change :+1: